tensorboard/plugins/hparams/api.py

```diff
@@ -412,3 +413,92 @@ def as_proto(self):
         description=self._description,
         dataset_type=self._dataset_type,
     )
+
+
+class KerasCallback(tf.keras.callbacks.Callback):
+  """Callback for logging hyperparameters to TensorBoard.
+
+  NOTE: This callback only works in TensorFlow eager mode.
+  """
+
+  def __init__(
+      self,
+      logdir,
+      hparams,
+      group_name=None,
+  ):
+    """Create a callback for logging hyperparameters to TensorBoard.
+
+    Each callback object is good for one session only.
+
+    Args:
+      logdir: The log directory for this session.
+      hparams: A `dict` mapping hyperparameters to the values used in
+        this session. Keys should be the names of `HParam` objects used
+        in an `Experiment`, or the `HParam` objects themselves. Values
+        should be Python `bool`, `int`, `float`, or `string` values,
+        depending on the type of the hyperparameter.
+      group_name: The name of the session group containing this session,
+        as a string or `None`. If `None` or empty, the group name is
+        taken to be the session ID.
+
+    Raises:
+      ValueError: If two entries in `hparams` share the same
+        hyperparameter name.
+    """
+    self._hparams = _normalize_hparams(hparams)
+    self._group_name = group_name if group_name is not None else ""
+    self._writer = tf.compat.v2.summary.create_file_writer(logdir)
+
+  def _write_summary(self, pb, step=None):
+    raw_pb = pb.SerializeToString()
+    with self._writer.as_default():
+      result = tf.compat.v2.summary.experimental.write_raw_pb(raw_pb, step=step)
+    sentinel = object()
+    if getattr(result, "numpy", sentinel) is sentinel:
```

Comment:
Any reason not just to use `tf.executing_eagerly()` up at the top?
https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/executing_eagerly